drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1161,9 +1161,9 @@ static void usbd_event_handler(nrfx_usbd_evt_t const *const p_event)
 		nrfx_usbd_setup_t drv_setup;
 
 		nrfx_usbd_setup_get(&drv_setup);
-		if ((drv_setup.bRequest != REQ_SET_ADDRESS)
-		    || (REQTYPE_GET_TYPE(drv_setup.bmRequestType)
-			!= REQTYPE_TYPE_STANDARD)) {
+		if ((drv_setup.bRequest != USB_REQ_SET_ADDRESS)
+		    || (USB_REQTYPE_GET_TYPE(drv_setup.bmRequestType)
+			!= USB_REQTYPE_TYPE_STANDARD)) {
```

Comment:
Wouldn't this work well?

``` diff
diff --git a/drivers/usb/device/usb_dc_nrfx.c b/drivers/usb/device/usb_dc_nrfx.c
index 2a7242fd18..c47ab416ee 100644
--- a/drivers/usb/device/usb_dc_nrfx.c
+++ b/drivers/usb/device/usb_dc_nrfx.c
@@ -1158,12 +1158,11 @@ static void usbd_event_handler(nrfx_usbd_evt_t const *const p_event)
 		break;
 
 	case NRFX_USBD_EVT_SETUP: {
-		nrfx_usbd_setup_t drv_setup;
+		struct usb_setup_packet setup;
 
-		nrfx_usbd_setup_get(&drv_setup);
-		if ((drv_setup.bRequest != USB_REQ_SET_ADDRESS)
-		    || (USB_REQTYPE_GET_TYPE(drv_setup.bmRequestType)
-			!= USB_REQTYPE_TYPE_STANDARD)) {
+		nrfx_usbd_setup_get((nrfx_usbd_setup_t *)&setup);
+		if ((setup.bRequest != USB_REQ_SET_ADDRESS)
+		    || (setup.RequestType.type != USB_REQTYPE_TYPE_STANDARD)) {
 			/* SetAddress is habdled by USBD hardware.
 			 * No software action required.
 			 */
```

Reply:
Maybe, but I do not know why it was not done same way as in usbd_work_process_setup(). Local USB_REQTYPE_GET_TYPE macro resolves it (temporary) very well and without any behavior change.